File: RecommendApp/backEnd/RecommendAPI/Helpers/UserParams.cs
using RecommendAPI.Entities.Enums;

namespace RecommendAPI.Helpers
{
    public class UserParams : PaginationParams
    {
        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;
        public string Username { get; set; }
        public string Name { get; set; }
        public string MidiaTitulo { get; set; }
        public MidiaTypeEnum MidiaTipo { get; set; }
        public bool Avaliado { get; set; } = true;

        public string OrderBy { get; set; } = "lastActive";
    }
}

File: RecommendApp/client/src/app/_models/userParams.ts
import { User } from './user';

export class UserParams {
    gender: string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    pageSize = 7;
    username:string= '';
    name:string= '';
    midiaTitulo:string = '';
    midiaTipo:string ='3';
    orderBy = 'lastActive';
    avaliado = 'true';

    constructor(user: User) {
        this.gender = user.gender === 'female' ? 'male' : 'female';
    }
}